dubbo-config/dubbo-config-api/src/main/java/org/apache/dubbo/config/ServiceConfig.java

```diff
@@ -724,4 +736,45 @@ public DubboBootstrap getBootstrap() {
     public void setBootstrap(DubboBootstrap bootstrap) {
         this.bootstrap = bootstrap;
     }
+
+    protected void delayExport() {
+        doExport();
+    }
+
+    private void duplicateExportCheck() {
+        String generateVersion = version;
+        String generateGroup = group;
+
+        if (StringUtils.isBlank(version) && provider != null) {
+            generateVersion = provider.getVersion();
+        }
+
+        if (StringUtils.isBlank(group) && provider != null) {
+            generateGroup = provider.getGroup();
+        }
+        StringBuilder stringBuilder = new StringBuilder();
+        stringBuilder.append("ServiceBean:");
+
+        if (!StringUtils.isBlank(generateGroup)) {
+            stringBuilder.append(generateGroup);
+        }
+
+        stringBuilder.append("/").append(interfaceName);
+
+        if (!StringUtils.isBlank(generateVersion)) {
+            stringBuilder.append(":").append(generateVersion);
+        }
+
+        serviceBeanId = stringBuilder.toString();
+
+        if (exportedServiceBeanIds.contains(serviceBeanId)) {
+            throw new IllegalArgumentException("The Duplicated BeanDefinition of ServiceBean[group:" +
```

Comment:
The error message should include the conflicting ServiceConfig instances (toString directly), which makes it easier to locate the sources of the problem.

Reply:
ok, thanks for your guidance